activitysim/abm/models/initialize_from_usim.py

```diff
@@ -130,46 +183,9 @@ def colleges(blocks):
 
 @orca.column('blocks', cache = True)
 def TAZ(blocks, zones):
-
-    # Tranform blocks to a Geopandas dataframe
-    blocks_df = blocks.to_frame(columns=['x', 'y'])
-    zones_df =  zones.to_frame(columns=['geometry', 'area'])
-    h3_gpd = gpd.GeoDataFrame(zones_df, crs='EPSG:4326')
-
-    blocks_df = gpd.GeoDataFrame(
-        blocks_df, geometry=gpd.points_from_xy(blocks_df.x, blocks_df.y),
-        crs="EPSG:4326")
-
-    # Spatial join 
-    blocks_df = gpd.sjoin(blocks_df, h3_gpd, how='left', op = 'intersects')
-
-    # Drop duplicates and keep the one with the smallest H3 area
-    blocks_df = blocks_df.sort_values('area')
-    blocks_df.drop_duplicates(subset = ['x', 'y'], keep = 'first', inplace = True) 
-    
-    # Buffer unassigned blocks until they reach a hexbin. 
-    null_blocks = blocks_df[blocks_df.index_right.isnull()].drop(columns = ['index_right','area'])
-
-    result_list = []
-    for index, block in null_blocks.iterrows():
-        buff_size = 0.0001
-        matched = False
-        geo_block = gpd.GeoDataFrame(block, crs='EPSG:4326').T
-        while matched == False:
-            geo_block.geometry = geo_block.geometry.buffer(buff_size)
-            result = gpd.sjoin(geo_block, h3_gpd, how = 'left', op = 'intersects')
-            matched = ~result.index_right.isnull()[0]
-            buff_size = buff_size + 0.0001
-        result_list.append(result.iloc[0:1])
-
-    null_blocks = pd.concat(result_list)
-    
-    # Concatenate newly assigned blocks to the main blocks table 
-    blocks_df = blocks_df.dropna()
-    blocks_df = pd.concat([blocks_df, null_blocks], axis = 0)
-    
-    return blocks_df.index_right
-
+    blocks_df = blocks.to_frame(columns = ['x', 'y'])
+    h3_gpd =  zones.to_frame(columns = ['geometry', 'area'])
```

Comment:
Even though `zones.to_frame()` might return a GeoDataFrame object type, it does not persist the crs attribute, so you should explicitly set it before passing it to `assign_taz()`

Reply:
great. I'll fixed the issue inside the `assign_taz()`.